lib/bottomnavigation/bottomnav.dart

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:SquanchyFlutter/eventdetails/eventdetails.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:SquanchyFlutter/favourites/favourites.dart';
+import 'package:SquanchyFlutter/schedule/schedule.dart';
+import 'package:SquanchyFlutter/venue/venuedetails.dart';
+
+class NavigationBar extends StatelessWidget {
+
+  static const String routeName = '/cupertino/navigation';
+
+  @override
+  Widget build(BuildContext context) {
+    var theme = Theme.of(context);
+    var tabBarTextStyle = new TextStyle(color: Colors.blueAccent, fontFamily: 'Quicksand');
+
+    var tabBar = new CupertinoTabBar(
+        activeColor: theme.accentIconTheme.color,
+        inactiveColor: theme.iconTheme.color,
+
+        items: [
+          new BottomNavigationBarItem(icon: new Image.asset('assets/ic_schedule_selected.png'), title: new Text("Schedule", style: tabBarTextStyle)),
+          new BottomNavigationBarItem(icon: new Image.asset('assets/ic_favourites_unselected.png'), title: new Text("Favourites", style: tabBarTextStyle)),
+          new BottomNavigationBarItem(icon: new Image.asset('assets/ic_venue_unselected.png'), title: new Text("Venue Details", style: tabBarTextStyle))
+        ]);
+
+
+    return new WillPopScope(
+      // Prevent swipe popping of this page. Use explicit exit buttons only.
+      onWillPop: () => new Future<bool>.value(true),
+      child: new CupertinoTabScaffold(
+        tabBar: tabBar,
+        tabBuilder: (BuildContext context, int index) {
+          return new DefaultTextStyle(
+            style: const TextStyle(
+              fontFamily: '.SF UI Text',
+              fontSize: 17.0,
+              color: Colors.blue,
+            ),
+            child: new CupertinoTabView(
+              builder: (BuildContext context) {
+                switch (index) {
+                  case 0:
+                    return new ScheduleView();
+                    break;
+                  case 1:
+                    return new FavouritesView();
+                    break;
+                  case 2:
+                    return new VenueDetailsView();
+                    break;
+                  default:
```

Comment:
should we throw here, in case we add a tab and forget to implement it?

on second thought we could have something like this (pseudo code cause I can't dart anymore):

```dart
var pages = [
  new Page(navItem, () => new ScheduleView())
];
```

where the first item is the one you put in `items`, the second is a function that returns a schedule view (`buildView()`?)

then when we need the items we could do:

```dart
items: pages.map(page => page.navItem)
```

and when we need to get the view we do:

```dart
child: new CupertinoTabView(
              builder: (BuildContext context) {
                return pages[index].buildView();
```

this way we keep the 1:1 on button:view and we remove the switch